### PR TITLE
amazon-ecs-cli: init at 1.12.1

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Motivation for this change
SInce we needed this for the deployment process I figured I push it upstream

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/virtualization/amazon-ecs-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      chmod +x $out/bin/ecs-cli
+    '';  # */
+
+  meta = with stdenv.lib; {
+    homepage = https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_CLI.html;
+    description = "The Amazon ECS command line interface";
+    longDescription = "The Amazon Elastic Container Service (Amazon ECS) command line interface (CLI) provides high-level commands to simplify creating, updating, and monitoring clusters and tasks from a local development environment.";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ Scriptkiddi ];
+    platforms = platforms.unix;
```
Comment (matthewbauer) [inline pkgs/tools/virtualization/amazon-ecs-cli/default.nix]: This should be ```platforms = [ "x86_64-linux" ];```

Comment (Scriptkiddi) [reply pkgs/tools/virtualization/amazon-ecs-cli/default.nix]: thanks, done

File: pkgs/tools/virtualization/amazon-ecs-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+stdenv.mkDerivation rec {
+  name = "amazon-ecs-cli";
+  version = "1.12.1";
+
+  src = fetchurl {
+    url = "https://s3.amazonaws.com/amazon-ecs-cli/ecs-cli-linux-amd64-v${version}";
+    sha256 = "100iv4cchnxl1s02higga5v3hvawi4c7sqva97x34qigr4r7fxwm";
+  };
+
+  unpackCmd = "mkdir bin; cp $src ./bin/ecs-cli";
```
Comment (joachifm) [inline pkgs/tools/virtualization/amazon-ecs-cli/default.nix]: Seems to me that `unpackCmd` and `installPhase` can be replaced with
```
install -D $src $out/bin/ecs-cli
```

File: pkgs/tools/virtualization/amazon-ecs-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchurl}:
+
+stdenv.mkDerivation rec {
+  name = "amazon-ecs-cli";
```
Comment (joachifm) [inline pkgs/tools/virtualization/amazon-ecs-cli/default.nix]: You need to add `version` to `name` explicitly

File: pkgs/tools/virtualization/amazon-ecs-cli/default.nix (hunk 2 above)
Comment (Scriptkiddi) [reply pkgs/tools/virtualization/amazon-ecs-cli/default.nix]: I have not found a way to disable the unpack phase, could you point me in the right direction?

Comment (joachifm) [reply pkgs/tools/virtualization/amazon-ecs-cli/default.nix]: A simple way to disable specific phases is to set them to ":", as in
```nix
{
   unpackPhase = ":";
}
```

Comment (Scriptkiddi) [reply pkgs/tools/virtualization/amazon-ecs-cli/default.nix]: like this?

Comment (joachifm) [reply pkgs/tools/virtualization/amazon-ecs-cli/default.nix]: LGTM.  Using `install` would shorten it further, as noted above. Even simpler would be
```nix
runCommand "foo" { src = fetchurl { /* ... */ }; } ''install -D $src $out/bin/foo''
```
There may even be a suitable canned builder that captures this pattern entirely. That said I think what you have is acceptable as is ...

**Pull request merged**